Code/GraphMol/Chirality.cpp

```diff
@@ -1089,6 +1088,88 @@ void buildCIPInvariants(const ROMol &mol, DOUBLE_VECT &res) {
   }
 }
 
+//! Lightweight sortable wrapper that references a CIP entry and keeps track of
+//! the current rank.
+struct SortableCIPReference {
+  SortableCIPReference(CIP_ENTRY* cipRef, const int atomIdx): cip(cipRef), atomIdx(atomIdx) {}
+  SortableCIPReference(SortableCIPReference && other) noexcept {
+    cip = other.cip;
+    atomIdx = other.atomIdx;
+    other.cip = nullptr;
+    currRank = other.currRank;
+  }
+  SortableCIPReference & operator=(SortableCIPReference && other) noexcept {
```

Comment:
Isn't it good practice to test for self-assignment here?

Reply:
Done!